### PR TITLE
HTSD-13 Add business rules around new Start and End date fields

### PR DESCRIPTION
**Pull request by rajanimohan22**
HTSD-13 Add business rules around new Start and End date fields

New feature

Adding business rules around new Start and End date fields of the Reporting Period of the ReportingEntity Tag element

 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/uk/gov/hmrc/cbcrfrontend/services/CBCBusinessRuleValidator.scala [hunk 1]
```diff
@@ -193,6 +193,12 @@ class CBCBusinessRuleValidator @Inject()(
       .leftMap(_ => InvalidXMLError("xmlValidationError.InvalidDate"))
       .toValidatedNel
 
+  private def extractEntityReportingPeriod1(in: RawReportingEntity): ValidBusinessResult[Boolean] =
```
Comment (calinursaciuc) [inline app/uk/gov/hmrc/cbcrfrontend/services/CBCBusinessRuleValidator.scala:196]: I suppose this to be deleted?

Comment (rajanimohan22) [reply app/uk/gov/hmrc/cbcrfrontend/services/CBCBusinessRuleValidator.scala:196]: Yes

File: app/uk/gov/hmrc/cbcrfrontend/services/XmlInfoExtract.scala [hunk 2]
```diff
@@ -168,6 +168,7 @@ class XmlInfoExtract {
       ds =>
         {
           val otherInfo = (ds \ "OtherInfo").text
+          val otherInfoLanguage = (ds \ "OtherInfo") \@ "language"
```
Comment (calinursaciuc) [inline app/uk/gov/hmrc/cbcrfrontend/services/XmlInfoExtract.scala:171]: I don't think we need to extract the language if we don't use it especially that might not be there.

File: app/uk/gov/hmrc/cbcrfrontend/services/CBCBusinessRuleValidator.scala [hunk 3]
```diff
@@ -265,6 +275,40 @@ class CBCBusinessRuleValidator @Inject()(
     if (xmlInfo.additionalInfo.forall(!_.otherInfo.trim.isEmpty)) xmlInfo.validNel
     else OtherInfoEmpty.invalidNel
 
+  private def validateStartDateBeforeEndDate(entity: ReportingEntity): ValidBusinessResult[ReportingEntity] =
+    if (entity.entityReportingPeriod.startDate.isAfter(entity.entityReportingPeriod.endDate))
```
[the remark this answers is not in the document]
Comment (rajanimohan22) [reply app/uk/gov/hmrc/cbcrfrontend/services/CBCBusinessRuleValidator.scala:273]: Ok, I think we need to add == also

[the remark this answers is not in the document]
Comment (rajanimohan22) [reply app/uk/gov/hmrc/cbcrfrontend/services/CBCBusinessRuleValidator.scala:273]: It should be at least a day before

File: app/uk/gov/hmrc/cbcrfrontend/services/XmlInfoExtract.scala (hunk 2 above)
Comment (rajanimohan22) [reply app/uk/gov/hmrc/cbcrfrontend/services/XmlInfoExtract.scala:171]: Oh this is a mistake, let me revet it.

**Review by calinursaciuc — APPROVED** [review]

**Pull request merged**